be/src/http/action/datacache_action.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "http/action/datacache_action.h"
+
+#include <fmt/format.h>
+#include <rapidjson/document.h>
+#include <rapidjson/prettywriter.h>
+#include <rapidjson/stringbuffer.h>
+
+#include <string>
+
+#include "block_cache/block_cache.h"
+#include "http/http_channel.h"
+#include "http/http_headers.h"
+#include "http/http_request.h"
+#include "http/http_status.h"
+
+namespace starrocks {
+
+const static std::string HEADER_JSON = "application/json";
+const static std::string ACTION_KEY = "action";
+const static std::string ACTION_STAT = "stat";
+const static std::string ACTION_INVALIDATE_ALL = "invalidate_all";
+
+std::string cache_status_str(const DataCacheStatus& status) {
+    std::string str_status;
+    switch (status) {
+    case DataCacheStatus::NORMAL:
+        str_status = "NORMAL";
+        break;
+    case DataCacheStatus::UPDATING:
+        str_status = "UPDATING";
+        break;
+    case DataCacheStatus::ABNORMAL:
+        str_status = "ABNORMAL";
+        break;
+    }
+    return str_status;
+}
+
+void DataCacheAction::handle(HttpRequest* req) {
+    VLOG_ROW << req->debug_string();
+    const auto& action = req->param(ACTION_KEY);
+    if (req->method() == HttpMethod::GET) {
+        if (action == ACTION_STAT) {
+            auto block_cache = _exec_env->block_cache();
+            if (!block_cache || !block_cache->is_initialized()) {
+                _handle_error(req, strings::Substitute("Cache system is not ready"));
+            } else {
+                _handle_stat(req, block_cache);
+            }
+        } else {
+            _handle_error(req, strings::Substitute("Not support GET method: '$0'", req->uri()));
+        }
+    } else {
+        _handle_error(req,
```

Comment:
usually if the method is not allowed, it should be a HTTP 405 error. not a HTTP 200 with response body `error: not supported ....`